digits/extensions/view/imageSegmentation/view.py

```diff
@@ -134,24 +165,66 @@ def process_data(self, input_id, input_data, output_data):
         """
         # assume the only output is a CHW image where C is the number
         # of classes, H and W are the height and width of the image
-        data = output_data[output_data.keys()[0]].astype('float32')
+        class_data = output_data[output_data.keys()[0]].astype('float32')
         # retain only the top class for each pixel
-        data = np.argmax(data,axis=0)
+        class_data = np.argmax(class_data, axis=0).astype('uint8')
 
         # remember the classes we found
-        found_classes = np.unique(data)
-        self.found_classes = np.unique(np.concatenate(
-            (self.found_classes, found_classes)))
+        found_classes = np.unique(class_data)
 
         # convert using color map (assume 8-bit output)
         if self.map:
-            data = self.map.to_rgba(data)*255
-            # keep RGB values only, remove alpha channel
-            data = data[:, :, 0:3]
-
-        # convert to uint8
-        data = data.astype('uint8')
-        # convert to PIL image
-        image = PIL.Image.fromarray(data)
-
-        return image
+            fill_data = (self.map.to_rgba(class_data)*255).astype('uint8')
+        else:
+            fill_data = np.ndarray((class_data.shape[0], class_data.shape[1], 4), dtype='uint8')
+            for x in xrange(3):
+                fill_data[:, :, x] = class_data.copy()
+
+        # Assuming that class 0 is the background
+        mask = np.greater(class_data, 0)
+        fill_data[:, :, 3] = mask * 255
+
+        # Black mask of non-segmented pixels
+        mask_data = np.zeros(fill_data.shape, dtype='uint8')
+        mask_data[:, :, 3] = (1 - mask) * 255
+
+        # Generate outlines around segmented classes
+        if len(found_classes) > 1:
+            # Assuming that class 0 is the background.
+            line_mask = np.zeros(class_data.shape, dtype=bool)
+            for c in (x for x in found_classes if x != 0):
+                c_mask = np.equal(class_data, c)
+                # Find the signed distance from the zero contour
+                distance = skfmm.distance(c_mask.astype('float32') - 0.5)
+                # Accumulate the mask for all classes
+                line_width = 3
+                line_mask |= c_mask & np.less(distance, line_width)
+
+        # add the outlines to the input image
+        for x in xrange(3):
+            input_data[:, :, x] = (input_data[:, :, x] * (1 - line_mask) +
```

Comment:
@jmancewicz yes, this was an image from the Sunnybrook dataset (with channel conversion = 'none')